src/components/LanguageOption.tsx

```diff
@@ -146,6 +146,8 @@ export default function LanguageOption(props: LanguageOptionProps) {
       .then(() => {
         const metadataExisted = Boolean(metadata?._createdAt)
 
+        callback?.(newTranslationDocument, newMetadataDocument, client)
```

Comment:
The `newMetadataDocument` only includes the currently created translation in the array of refs here - it's only used in the `createIfNotExists` transaction. If it does exist then the `metadataPatch` is the important thing.

I think maybe pass the `metadataId` instead here. If somebody needs to use this document, they could query it with the client.

I'm also wondering if it'd be helpful to pass the source and destination language IDs here too. I could see them being useful in callback function (albeit they are available in the documents, but the field name could vary and potentially change).

If more props are added, these maybe should be passed as an object rather than individual props.

Reply:
Great input, will fix!